test/cpp/end2end/end2end_test.cc

```diff
@@ -196,16 +196,18 @@ class TestServiceImplDupPkg
 class TestScenario {
  public:
   TestScenario(bool interceptors, bool proxy, bool inproc_stub,
-               const grpc::string& creds_type)
+               const grpc::string& creds_type, bool use_callback_server)
```

Comment:
Nit; this is getting messy with so many params

at some point this should be refactored to take in a struct like `ScenarioConfig`. At the least add a todo for this? non blocking for this PR

Reply:
Sure.